torch_geometric/loader/link_neighbor_loader.py

```diff
@@ -21,14 +23,28 @@ def __init__(
         self,
         data,
         *args,
+        edge_time=None,
         neg_sampling_ratio: float = 0.0,
         num_src_nodes: Optional[int] = None,
         num_dst_nodes: Optional[int] = None,
         **kwargs,
     ):
         super().__init__(data, *args, **kwargs)
         self.neg_sampling_ratio = neg_sampling_ratio
-
+        self.edge_time = edge_time
+
+        # TODO if self.edge_time is not None and
+        # input_type doesn't have time attribute
+        # set it to float(inf).
```

Comment:
This won't work as `node_time_dict` is of type integer.

Reply:
Good point, can we set it to the integer representable `torch.long` then?

Reply:
I think we should just remove the TODO here and expect that `node_time_dict` is present in case `edge_label_time` is present (and vice versa).

Reply:
`node_time_dict` needs to exist, but there might be cases where `src` and `dst` nodes might not have a `time_attr` *i.e* `node_time_dict[input_type[0]]` might not exist.

Reply:
Ah, I see, then I misread. In that case we don't want to fill the `node_time_dict` of that node type with any temporal information, right?

Reply:
Well with this hacky implementation we do, because the `edge_label_time` is "passed" to the `src`(input_type[0]) and `dst`(input_type[-1]) nodes which is then passed to the temporal sampler.

Reply:
got it. You are right, we cannot simply not set it.